var/spack/repos/builtin/packages/python-venv/package.py

```diff
@@ -93,6 +93,17 @@ def setup_dependent_run_environment(self, env, dependent_spec):
             path = os.path.join(dependent_spec.prefix, directory)
             if os.path.isdir(path):
                 env.prepend_path("PYTHONPATH", path)
+        try:
+            dep_bin_dir = dependent_spec.package.bindir
+            if os.path.isdir(dep_bin_dir):
+                env.prepend_path("PATH", dep_bin_dir)
+        except AttributeError:
```

Comment:
Can you just use `hasattr` instead of catching a generic exception?

Reply:
Sounds good!